src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java

```diff
@@ -0,0 +1,128 @@
+package io.github.reconsolidated.zpibackend.features.reservation;
+
+import io.github.reconsolidated.zpibackend.utils.BooleanListToStringConverter;
+import lombok.*;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class ScheduleSlot {
+    @Id
+    @GeneratedValue(generator = "schedule_slot_generator")
+    private Long scheduleSlotId;
+    private LocalDateTime startDateTime;
+    private LocalDateTime endDateTime;
+    private Integer amount;
+    @Convert(converter = BooleanListToStringConverter.class)
+    private List<Boolean> itemsAvailability;
+    private ReservationType type = ReservationType.NONE;
+
+    public ScheduleSlot(LocalDateTime startDateTime, LocalDateTime endDateTime, Integer amount) {
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+        this.amount = amount;
+        this.itemsAvailability = new ArrayList<>(amount);
+        for(int i = 0; i < amount; i++) {
+            itemsAvailability.add(true);
+        }
+    }
+
+    public ScheduleSlot(LocalDateTime startDateTime, LocalDateTime endDateTime, Integer amount, ReservationType type) {
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+        this.type = type;
+        this.amount = amount;
+        this.itemsAvailability = new ArrayList<>(amount);
+        for(int i = 0; i < amount; i++) {
+            itemsAvailability.add(true);
+        }
+    }
+
+    public String toString(){
+        return startDateTime.toString() + "-" + endDateTime.toString() + ": " + this.amount;
+    }
+
+    public ScheduleSlot[] split(LocalDateTime splitBy) {
+        if(!startDateTime.isBefore(splitBy) || !endDateTime.isAfter(splitBy)) {
+            throw new IllegalArgumentException();
```

Comment:
add an explanation in the exception message

Reply:
description added